### PR TITLE
SuppplierOrder

### PR DESCRIPTION
**Pull request by SorinPopteanu**
I created the table for SupplierOrders. Unlike the table in ClientOrders, I've incorporated a TableView into a StackPane instead of a ScrollPane and haven't noticed any difference; if necessary, StackPane automatically creates its scrollbar. We'll see which option is better once we implement pagination. I still have some minor adjustments to make (for example, fetching the order status).

### CODE REVIEW

File: src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java [hunk 1]
```diff
@@ -82,12 +88,49 @@ private List<ClientOrder> handleOrdersResponse(Optional<List<ClientOrder>> order
...
             System.out.println("Orders received: " + clientOrders);
+            setTableView();
             bindDataToTableView();
+            setEditEvents();
         });
 
         return clientOrders;
     }
 
+    private void setTableView() {
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java]: configureTableView might be better naming

File: src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java [hunk 2]
```diff
@@ -82,12 +88,49 @@ private List<ClientOrder> handleOrdersResponse(Optional<List<ClientOrder>> order
...
+        tableView.setEditable(true);
+        clientIdColumn.setCellFactory(TextFieldTableCell.forTableColumn(new IntegerStringConverter()));
+        productIdColumn.setCellFactory(TextFieldTableCell.forTableColumn(new IntegerStringConverter()));
+        quantityColumn.setCellFactory(TextFieldTableCell.forTableColumn(new FloatStringConverter()));
+        statusColumn.setCellFactory(TextFieldTableCell.forTableColumn());
+
+        tableView.setMaxHeight(Double.MAX_VALUE);
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java:122]: You should identify the configuration that can be taken out of here and reused for the other tables we'll have, perhaps through a static method of a util class.

File: src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java [hunk 3]
```diff
@@ -59,9 +68,6 @@ public ClientOrdersController(FallbackManager fallbackManager,
     @Override
     public void setData(Client client) {
         this.client = client;
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java:77]: You should load the FallbackManagerView here and place it in a container with parent StackPane like I did throughout the app. Also make sure to reset fallbackManager before starting to load orders.

File: src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java [hunk 4]
```diff
@@ -82,12 +88,49 @@ private List<ClientOrder> handleOrdersResponse(Optional<List<ClientOrder>> order
...
+
+        // Set the column resize policy and their minimum width
+        tableView.setColumnResizePolicy(new Callback<TableView.ResizeFeatures, Boolean>() {
+            @Override
+            public Boolean call(TableView.ResizeFeatures param) {
+                return TableView.CONSTRAINED_RESIZE_POLICY.call(param) || Boolean.TRUE;
+            }
+        });
+
+//        tableView.addEventFilter(ScrollEvent.SCROLL, event -> {
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java]: Remove this if no longer needed

File: src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java [hunk 5]
```diff
@@ -98,9 +141,37 @@ private void bindDataToTableView() {
         estimatedDeliveryDateColumn.setCellValueFactory(new PropertyValueFactory<>("estimatedDeliveryDate"));
         deliveryDateColumn.setCellValueFactory(new PropertyValueFactory<>("deliveryDate"));
 
-        // Set the items in the table
         tableView.getItems().setAll(clientOrders);
+    }
+
+    private void setEditEvents() {
+        clientIdColumn.setOnEditCommit(event -> {
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java:147]: Generialize this setOnEditCommit block to not repeat yourself. But more importantly, do not go to the database for every edited value, there should be a save button somewhere (perhaps restricting to updating one order at a time).

File: src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java [hunk 6]
```diff
@@ -98,9 +141,37 @@ private void bindDataToTableView() {
...
+        });
+
+        statusColumn.setOnEditCommit(event -> {
+            ClientOrder order = event.getRowValue();
+            order.setStatus(event.getNewValue());
+            updateInDatabase(order);
+        });
+    }
 
+    private void updateInDatabase(ClientOrder order) {
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/client/controller/ClientOrdersController.java:172]: Make the update service method and use it here to update. I'll add a Toast system later for confirming successful operation/signaling error.

File: src/main/java/org/chainoptim/desktop/features/supplier/controller/SupplierOrdersController.java [hunk 7]
```diff
@@ -1,11 +1,153 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.cell.PropertyValueFactory;
+import javafx.scene.control.cell.TextFieldTableCell;
+import javafx.scene.layout.StackPane;
+import javafx.util.Callback;
+import javafx.util.converter.FloatStringConverter;
+import javafx.util.converter.IntegerStringConverter;
+import org.chainoptim.desktop.features.client.model.ClientOrder;
```
Comment (TudorOrban) [inline src/main/java/org/chainoptim/desktop/features/supplier/controller/SupplierOrdersController.java]: Ensure Intellij places org.chainoptim.desktop dependencies before external ones for consistency. Go to:
Settings > Editor > Code Style > Java > Imports 
and under Layout static imports separately add org.chainoptim.desktop and move it to the top.

File: src/main/resources/css/scrollbar.css [hunk 8]
```diff
@@ -0,0 +1,70 @@
+.scroll-pane {
```
Comment (TudorOrban) [inline src/main/resources/css/scrollbar.css:1]: Why are you using decimals in this stylesheet?

File: src/main/resources/org/chainoptim/desktop/features/client/ClientOrdersView.fxml [hunk 9]
```diff
@@ -4,23 +4,23 @@
...
-        <TableColumn text="Product ID" fx:id="productIdColumn"/>
-        <TableColumn text="Quantity" fx:id="quantityColumn"/>
-        <TableColumn text="Status" fx:id="statusColumn"/>
-        <TableColumn text="Order Date" fx:id="orderDateColumn"/>
-        <TableColumn text="Estimated Delivery Date" fx:id="estimatedDeliveryDateColumn"/>
-        <TableColumn text="Delivery Date" fx:id="deliveryDateColumn"/>
-
-        </columns>
-    </TableView>
+    <ScrollPane fx:id="scrollPane" styleClass="scroll-pane" fitToWidth="true" AnchorPane.topAnchor="0.0" AnchorPane.leftAnchor="0.0" AnchorPane.rightAnchor="0.0" AnchorPane.bottomAnchor="40.0">
```
Comment (TudorOrban) [inline src/main/resources/org/chainoptim/desktop/features/client/ClientOrdersView.fxml]: Decide on either StackPane and ScrollPane for consistency. Also not sure if that outer AnchorPane is necessary.

File: src/main/resources/css/scrollbar.css (hunk 8 above)
Comment (SorinPopteanu) [reply src/main/resources/css/scrollbar.css:1]: In Stack Overflow, I saw that they were using decimals, and I thought maybe that's the practice. I removed them.

**Pull request merged**